apiserver/src/main/java/org/dependencytrack/integrations/gitlab/GitLabRole.java

```diff
@@ -0,0 +1,103 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) OWASP Foundation. All Rights Reserved.
+ */
+package org.dependencytrack.integrations.gitlab;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.dependencytrack.auth.Permissions;
+
+/**
+ * Definitions of access levels/roles as defined by GitLab.
+ */
+public enum GitLabRole {
+
+    GUEST(10, "GitLab Project Guest", Set.of( // Applies to private and internal projects only
+            Permissions.Constants.VIEW_PORTFOLIO,
+            Permissions.Constants.VIEW_VULNERABILITY,
+            Permissions.Constants.VIEW_BADGES)),
+    PLANNER(15, "GitLab Project Planner", Set.of(
+            Permissions.Constants.VIEW_POLICY_VIOLATION)),
+    REPORTER(20, "GitLab Project Reporter", Set.of(
+            Permissions.Constants.VIEW_POLICY_VIOLATION)),
+    DEVELOPER(30, "GitLab Project Developer", Set.of(
+            Permissions.Constants.BOM_UPLOAD,
+            Permissions.Constants.PORTFOLIO_MANAGEMENT_READ,
+            Permissions.Constants.VULNERABILITY_ANALYSIS_READ,
+            Permissions.Constants.PROJECT_CREATION_UPLOAD)),
+    MAINTAINER(40, "GitLab Project Maintainer", Set.of(
+            Permissions.Constants.PORTFOLIO_MANAGEMENT,
+            Permissions.Constants.PORTFOLIO_MANAGEMENT_CREATE,
+            Permissions.Constants.PORTFOLIO_MANAGEMENT_UPDATE,
+            Permissions.Constants.PORTFOLIO_MANAGEMENT_DELETE,
+            Permissions.Constants.VULNERABILITY_ANALYSIS,
+            Permissions.Constants.VULNERABILITY_ANALYSIS_CREATE,
+            Permissions.Constants.VULNERABILITY_ANALYSIS_UPDATE,
+            Permissions.Constants.POLICY_MANAGEMENT,
+            Permissions.Constants.POLICY_MANAGEMENT_CREATE,
+            Permissions.Constants.POLICY_MANAGEMENT_READ,
+            Permissions.Constants.POLICY_MANAGEMENT_UPDATE,
+            Permissions.Constants.POLICY_MANAGEMENT_DELETE)),
+    OWNER(50, "GitLab Project Owner", Set.of(
+            Permissions.Constants.ACCESS_MANAGEMENT,
+            Permissions.Constants.ACCESS_MANAGEMENT_CREATE,
+            Permissions.Constants.ACCESS_MANAGEMENT_READ,
+            Permissions.Constants.ACCESS_MANAGEMENT_UPDATE,
+            Permissions.Constants.ACCESS_MANAGEMENT_DELETE,
+            Permissions.Constants.SYSTEM_CONFIGURATION,
+            Permissions.Constants.SYSTEM_CONFIGURATION_CREATE,
+            Permissions.Constants.SYSTEM_CONFIGURATION_READ,
+            Permissions.Constants.SYSTEM_CONFIGURATION_UPDATE,
+            Permissions.Constants.SYSTEM_CONFIGURATION_DELETE,
+            Permissions.Constants.TAG_MANAGEMENT,
+            Permissions.Constants.TAG_MANAGEMENT_DELETE));
```

Comment:
These permissions are currently not scoped to projects, but apply to the entire system. This would make all project owners also system administrators, which is not what we want.